test/e2e/network/endpointslice.go

```diff
@@ -303,6 +304,131 @@ var _ = SIGDescribe("EndpointSlice", func() {
+				Containers: []v1.Container{
+					{
+						Name:    "container1",
+						Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+						Command: []string{"sleep", "1000000"},
+						Ports: []v1.ContainerPort{{
+							Name:          "example-name",
+							ContainerPort: int32(3000),
+						}},
+					},
+				},
+			},
+		})
+
+		svc1 := createServiceReportErr(cs, f.Namespace.Name, &v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "example-int-port",
+			},
+			Spec: v1.ServiceSpec{
+				Selector: map[string]string{labelPod1: labelValue},
+				Ports: []v1.ServicePort{{
+					Name:       "example",
+					Port:       80,
+					TargetPort: intstr.FromInt(3000),
+					Protocol:   v1.ProtocolTCP,
+				}},
+			},
+		})
+
+		// wait for pod to start up
+		err := wait.Poll(5*time.Second, time.Minute, func() (bool, error) {
+			var err error
+			pod1, err = podClient.Get(context.TODO(), pod1.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			if len(pod1.Status.PodIPs) == 0 {
+				return false, nil
+			}
+
+			return true, nil
+		})
+		framework.ExpectNoError(err, "timed out waiting for Pods to have IPs assigned")
+
+		ginkgo.By("referencing a single matching pod")
+		expectEndpointsAndSlices(cs, f.Namespace.Name, svc1, []*v1.Pod{pod1}, 1, 1, false)
+
+		// Delete pod1 with long termination grace period.
+		// NOTE: for this test we need to use a pod that ignores SIGTERM from kubelet
+		// in order to keep terminating pods around long enough to test this behavior.
+		err = podClient.Delete(context.TODO(), pod1.Name, metav1.DeleteOptions{})
+		framework.ExpectNoError(err, fmt.Sprintf("failed to delete pod %s/%s", pod1.Namespace, pod1.Name))
+
+		// ensure deletion timestamp
+		err = wait.Poll(5*time.Second, time.Minute, func() (bool, error) {
+			var err error
+			pod1, err = podClient.Get(context.TODO(), pod1.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+
+			if pod1.DeletionTimestamp == nil {
+				return false, nil
+			}
+
+			return true, nil
+		})
+		framework.ExpectNoError(err, "timed out waiting for pod to have deletion timestamp")
+
+		// TODO: update this test to include terminating endpoints once EndpointSliceTerminatingCondition is enabled by default
+		ginkgo.By("checking terminating endpoints are NOT included in EndpointSlice")
+		err = wait.Poll(2*time.Second, 10*time.Second, func() (bool, error) {
+			listOptions := metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", discoveryv1beta1.LabelServiceName, svc1.Name)}
+			esList, err := cs.DiscoveryV1beta1().EndpointSlices(f.Namespace.Name).List(context.TODO(), listOptions)
+			if err != nil {
+				return false, err
+			}
+			framework.ExpectNoError(err, fmt.Sprintf("error fetching EndpointSlice for Service %s/%s", f.Namespace.Name, svc1.Name))
+
+			numEndpoints := 0
+			for _, endpointSlice := range esList.Items {
+				numEndpoints += len(endpointSlice.Endpoints)
+			}
```

Comment:
We need @robscott here, I remember there are issues with e2e tests and leftovers with slices, he can advice on best practices to assert these conditions

Reply:
In this test we only care if the `numEndpoints` is 0 or not since we're testing to ensure terminating pods are never included in EndpointSlice. Even if we obtain a stale slice, I _think_ we should consider that a bug/failure since clients could end up consuming that.